ibm/data_source_ibm_satellite_host_script.go

```diff
@@ -0,0 +1,131 @@
+package ibm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path/filepath"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	homedir "github.com/mitchellh/go-homedir"
+	"github.ibm.com/ibmcloud/kubernetesservice-go-sdk/kubernetesserviceapiv1"
+)
+
+func dataSourceIBMSatelliteAttachHostScript() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceIBMSatelliteAttachHostScriptRead,
+
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "A unique name for the new Satellite location",
+			},
+			"labels": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         schema.HashString,
+				Description: "List of labels for the attach host",
+			},
+			"host_provider": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"script_dir": {
+				Description: "The directory where the satellite attach host script to be downloaded. Default is home directory ",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			"host_script": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Attach host script content",
+			},
+		},
+	}
+}
+
+func dataSourceIBMSatelliteAttachHostScriptRead(d *schema.ResourceData, meta interface{}) error {
+	location := d.Get("location").(string)
+	hostProvider := d.Get("host_provider").(string)
+	scriptDir := d.Get("script_dir").(string)
```

Comment:
No default set in schema..
Since it is optional it is advisable to have d.GetOK()

Reply:
added d.GetOK() for 'script_dir'